salt/modules/rabbitmqadmin.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+'''
+Module to provide RabbitMQAdmin compatibility to Salt.
+Todo: .
+'''
+from __future__ import absolute_import
+
+# Import salt libs
+import salt.utils
+
+# Import python libs
+import logging
+import json
+
+from rabbitmq import _format_response
+
+log = logging.getLogger(__name__)
+
+
+def translate_boolean(v):
+    if v:
+        return 'true'
+    else:
+        return 'false'
+
+
+def __virtual__():
+    '''
+    Only load if RabbitMQ rabbitmqcadmin is installed.
+    '''
+    return salt.utils.which('rabbitmqadmin') is not None
+
+
+def declare_queue(name, vhost, durable, auto_delete, runas=None):
+    '''
+    Adds a queue via rabbitmqadmin declare queue.
+
+    ./rabbitmqadmin declare queue --vhost=Some_Virtual_Host name=some_outgoing_queue durable=True auto_delete=False
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' rabbitmq declare_queue '<queue_name>' '<vhost_name>' True
+
+
+    Setup:
+    rabbitmqadmin is from http://localhost:15672/cli/rabbitmqadmin it needs to be in the path.
+    see rabbitmq-formula/rabbitmq/rabbit-management.sls
+
+    '''
+
+    durable = translate_boolean(durable)
+    auto_delete = translate_boolean(auto_delete)
+
+    if runas is None:
+        runas = salt.utils.get_user()
+    res = __salt__['cmd.run']('rabbitmqadmin declare queue --vhost={0} name={1} durable={2} auto_delete={3}'.format(
+        vhost, name, durable, auto_delete),
+                              python_shell=False, runas=runas)
+    log.debug(res)
+    msg = 'Declared'
+    return _format_response(res, msg)
+
+
+def declare_binding(source, vhost, destination, destination_type, routing_key, runas=None):
+    '''
+    Adds a exchange via rabbitmqctl declare binding
+
+    rabbitmqadmin declare binding --vhost=Some_Virtual_Host source=bloa destination=bla destination_type=queue routing_key=""
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' rabbitmq declare_exchange 'blah' '<vhost_name>' 'boo' "queue" ""
+
+    Setup:
+
+    rabbitmqadmin is from http://localhost:15672/cli/rabbitmqadmin
+    see rabbitmq-formula/rabbitmq/rabbit-management.sls
+
+    '''
+
+    if runas is None:
+        runas = salt.utils.get_user()
+    res = __salt__['cmd.run']('rabbitmqadmin declare binding --vhost={0} source={1} destination={2} destination_type={3} routing_key={4}'.format(vhost, source, destination, destination_type, routing_key),
+                              python_shell=False,
+                              runas=runas)
+    log.debug(res)
+    msg = 'Declared'
```

Comment:
We don't need this assignment. It can just be passed in as an arg on the following line.